framework/connection.py

```diff
@@ -0,0 +1,75 @@
+"""
+Module to connect to remote host
+"""
+
+import logging
+
+from common_ci_utils.connection import Connection
+from framework import config
+from paramiko.auth_handler import AuthenticationException, SSHException
+
+log = logging.getLogger(__name__)
+
+
+class SSHConnection:
```

Comment:
I think we should use singleton pattern:
https://refactoring.guru/design-patterns/singleton/python/example

Reply:
done